middle_end/flambda/un_anf.ml

```diff
@@ -122,32 +122,32 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
   let rec loop ~depth ~environment_vars : Clambda.ulambda -> unit = function
     (* No underscores in the pattern match, to reduce the chance of failing
        to traverse some subexpression. *)
-    | Uvar var -> add_use t var depth
+    | Uvar var -> add_use t var depth; environment_vars
     | Uconst const ->
       (* The only variables that might occur in [const] are those in constant
          closures---and those are all bound by such closures.  It follows that
          [const] cannot contain any variables that are bound in the current
          scope, so we do not need to count them here.  (The function bodies
          of the closures will be traversed when this function is called from
          [Flambda_to_clambda.to_clambda_closed_set_of_closures].) *)
-      ignore_uconstant const
+      ignore_uconstant const; environment_vars
     | Udirect_apply (label, args, dbg) ->
       ignore_function_label label;
-      List.iter (loop ~depth ~environment_vars) args;
-      ignore_debuginfo dbg
+      List.fold_left (loop ~depth ~environment_vars) args;
+      ignore_debuginfo dbg; environment_vars
     | Ugeneric_apply (func, args, dbg) ->
       loop ~depth ~environment_vars func;
-      List.iter (loop ~depth ~environment_vars) args;
-      ignore_debuginfo dbg
+      List.fold_left (loop ~depth ~environment_vars) args;
+      ignore_debuginfo dbg; environment_vars
     | Uclosure (functions, captured_variables) ->
-      List.iter (loop ~depth ~environment_vars) captured_variables;
+      List.fold_left (loop ~depth ~environment_vars) captured_variables;
       List.iter (fun (
```

Comment:
This `List.iter` also needs to be replaced by a `List.fold_left` call. This one is actually easier than most other cases, since the function argument to `List.iter` was already an explicit function definition. You just need to add the "accumulator" argument that `fold_left` expects (`environment_vars`, in this case) between the `fun` keyword and the existing argument (the whole `({ Clambda. label; arity; params; return; body; dbg; env; } as clos)` is a single argument, this syntax allows to give names to both the argument, `clos`, and all of its fields).
You will also need to return the new `environment_vars` at the end of this function (the end of this function is at the right parenthesis that matches the left parenthesis just before the `fun` keyword).
And like all other occurrences of `fold_left`, you will need to give a name to the result so that you can return it later.